src/core/taylor.rkt

```diff
@@ -4,29 +4,36 @@
 (require "../utils/common.rkt"
          "programs.rkt"
          "reduce.rkt"
-         "../syntax/syntax.rkt")
-(provide approximate)
-
-(define (approximate expr var #:transform [tform (cons identity identity)] #:iters [iters 5])
-  (define expr* (simplify (replace-expression expr var ((car tform) var))))
-  (match-define (cons offset coeffs) (taylor var expr*))
+         "../syntax/syntax.rkt"
+         "batch.rkt")
 
-  (define i 0)
-  (define terms '())
-
-  (define (next [iter 0])
-    (define coeff (simplify (replace-expression (coeffs i) var ((cdr tform) var))))
-    (set! i (+ i 1))
-    (match coeff
-      [0
-       (if (< iter iters)
-           (next (+ iter 1))
-           (simplify (make-horner ((cdr tform) var) (reverse terms))))]
-      [_
-       (set! terms (cons (cons coeff (- i offset 1)) terms))
-       (simplify (make-horner ((cdr tform) var) (reverse terms)))]))
+(provide approximate)
 
-  next)
+(define (approximate exprs var #:transform [tform (cons identity identity)] #:iters [iters 5])
+  (define exprs*
+    (for/list ([expr (in-list exprs)])
+      (simplify (replace-expression expr var ((car tform) var)))))
```

Comment:
Ok, so this is also interesting—`replace-expression` could definitely be batchified, at least in the really simple form used here. `simplify` seems tough but maybe could be too.

Reply:
Yeah, I have also looked at it, `simplify` would require a lot of effort, while `replace-expressions` is 5mins of rewriting.